contracts/libs/Authenticity.sol

```diff
@@ -20,7 +20,11 @@ library Authenticity {
     ) internal view returns (Types.Result) {
         uint256 size = txs.transferSize();
         uint256[2][] memory messages = new uint256[2][](size);
-        for (uint256 i = 0; i < size; i++) {
+        uint256[] memory senders = new uint256[](size);
+        // We reverse loop the transactions to compute nonce correctly
+        for (uint256 j = 0; j < size; j++) {
```

Comment:
TIL there's no `for (uint256 i = size - 1; i <=0; i --) {` because the i will overflow 😈😈😈